api/src/test/java/org/openmrs/module/locationbasedaccess/aop/UserSearchAdviserTest.java

```diff
@@ -0,0 +1,120 @@
+package org.openmrs.module.locationbasedaccess.aop;
+
+import java.util.List;
+import org.junit.Test;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+import org.junit.Before;
+import org.openmrs.Person;
+import org.openmrs.User;
+import org.openmrs.api.PersonService;
+import org.openmrs.api.UserService;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.locationbasedaccess.aop.common.AOPContextSensitiveTest;
+import org.openmrs.module.locationbasedaccess.aop.common.TestWithAOP;
+import org.openmrs.module.locationbasedaccess.aop.interceptor.UserServiceInterceptorAdvice;
+
+public class UserSearchAdviserTest extends AOPContextSensitiveTest {
+    
+    private UserService userService;
+    private PersonService personService;
+    
+    private static final String XML_FILENAME_WITH_ADMIN_USER_DATA = "include/AdminUserData.xml";
+ 
+    @Override
+    protected void setInterceptorAndServices(TestWithAOP testCase) {
+        testCase.setInterceptor(UserServiceInterceptorAdvice.class);
+        testCase.addService(UserService.class);
+    }
+    
+    public UserSearchAdviserTest() {
+    }
+
+    @Before
+    public void setUp()  throws Exception {
+        userService = Context.getUserService();
+        personService = Context.getPersonService();
+        executeDataSet(XML_FILENAME_WITH_ADMIN_USER_DATA);
+    }
+        
+    @Test
+    public void getUsers_adminCanAccessUsers(){

```

Comment:
By this test case, you didn't achieve that Admin can access users from different locations. Please look into the sample test case carefully, and understand the need of each and every line of that code.

Reply:
@suthagar23 I'm unable to find any location attribute for users like they is one for patients.

Reply:
Hey @suthagar23 is there any UserTestData.xml like there is for patients (PatientTestData.xml)

Reply:
You can create it :smile:, Or just explore around OpenMRS-Core repository. You can find something related to your requirements. Please add only required fields to the XML file

Reply:
Got it :smile: